plugin.py

```diff
@@ -46,77 +45,83 @@ def prepare(self):
         self.module_input = self.board_prefix + '.dsn'
         self.module_output = self.board_prefix + '.ses'
         
+        self.module_command = [self.java_path, "-jar", self.module_path, "-de", self.module_input, "-s"]
+        
         if os.path.isfile(self.module_input):
             os.remove(self.module_input)
         if os.path.isfile(self.module_output):
             os.remove(self.module_output)
-
+        
     # export board.dsn file from pcbnew
     def RunExport(self):
         ok = pcbnew.ExportSpecctraDSN(self.module_input)
         if ok and os.path.isfile(self.module_input):
             return True
         else:
-            show_error("""
+            wx_show_error("""
             Failed to invoke:
             * pcbnew.ExportSpecctraDSN
             """)
             return False
-        
+
     # auto route by invoking FreeRouting.jar
     def RunRouter(self):
-        command = [self.java_path, "-jar", self.module_path, "-de", self.module_input, "-s"]
-        try:
-            process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            output, error = process.communicate()
-            if process.returncode == 0:
+        command = self.module_command
+        dialog = wx_make_cancel("""
+        Complete or Terminate FreeRouting:
+        * to complete, close Java window
+        * to terminate, press Terminate here
+        """)
+        invoker = ProcessThread(command, dialog)
+        dialog.Show()  # dialog first
```

Comment:
Why is this needed?

Reply:
to avoid race condition, when `ProcessThread` is trying to destroy dialog which is not yet shown